04-onion-routing.md

```diff
@@ -601,6 +606,20 @@ If the `cltv-expiry` is too low, the final node MUST fail the HTLC:
 
 1. type: 17 (`final_expiry_too_soon`)
 
+If the `outgoing_cltv_value` does not match the `ctlv-expiry` of the
+HTLC at the final hop:
+
+1. type: 18 (`final_incorrect_cltv_expiry`)
+2. data:
+   * [4:cltv-expiry]
+
+If the `amt_to_forward` does not match the `incoming_htlc_amt` of
```

Comment:
Huh? The sender does know how much should be left for the last hop, so she can just set the fee to that value. Really it should always be `incoming_htlc_amt = fee + amt_to_forward`, so it is indifferent whether we have 0 `amt_to_forward` or 0 `fee`.